pkg/logs/sender/message_buffer.go

```diff
@@ -0,0 +1,77 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-2019 Datadog, Inc.
+
+package sender
+
+import (
+	"github.com/DataDog/datadog-agent/pkg/logs/message"
+)
+
+// MessageBuffer accumulates lines in buffer escaping all '\n'
+// and accumulates the total number of bytes of all lines in line representation (line + '\n') in rawDataLen
+type MessageBuffer struct {
+	maxBatchCount  int
+	maxRequestSize int
+	messageBuffer  []*message.Message
```

Comment:
there is no need to store the message buffer just to track it's content length, we could just use a counter instead.